hedera-node/src/main/java/com/hedera/services/store/contracts/AbstractStackedLedgerUpdater.java

```diff
@@ -107,7 +107,10 @@ public void commit() {
 				wrapped.updatedAccounts.put(mutable.getAddress(), mutable);
 			}
 			mutable.setNonce(updatedAccount.getNonce());
-			mutable.setBalance(updatedAccount.getBalance());
+
+			if(updatedAccount.getWrappedAccountNonce()>=0) {
```

Comment:
This is a rather mysterious `if` 😁 ...the implementation of `getWrappedAccountNonce()` is,
```
	public long getWrappedAccountNonce() {
		return account != null ? account.getNonce() : 0;
	}
```
...so as far as I can tell, the `updatedAccount.getWrappedAccountNonce() >= 0` condition is always true, and we always update the balance...but then why was this added?

Reply:
We added `WorldStateTokenAccount` which is returned from `HederaWorldState.get()` if we call `get` on a token address. `WorldStateTokenAccount`s have a -1 nonce, that's how we distinguish them from `normal` accounts.
This was necessary because setting the balance on a token was breaking the logic (I think `trackingAccounts` throws missing account exception)